src/main/java/slimeknights/tconstruct/shared/block/BlockTable.java

```diff
@@ -169,18 +175,27 @@ private void writeDataOntoItemstack(@Nonnull ItemStack item, @Nonnull IBlockAcce
     TileEntity te = world.getTileEntity(pos);
     if(te != null && te instanceof TileTable) {
       TileTable table = (TileTable) te;
+      NBTTagCompound tag = TagUtil.getTagSafe(item);
+
+      // texture
       NBTTagCompound data = table.getTextureBlock();
 
-      NBTTagCompound tag = TagUtil.getTagSafe(item);
-      tag.setTag(TileTable.FEET_TAG, data);
-      item.setTagCompound(tag);
+      if (!data.hasNoTags()) {
+        tag.setTag(TileTable.FEET_TAG, data);
+      }
 
-      // save inventory?
+      // save inventory, if not empty
       if(noInventorySave && keepInventory(state)) {
-        NBTTagCompound inventoryTag = new NBTTagCompound();
-        table.writeInventoryToNBT(inventoryTag);
-        tag.setTag("inventory", inventoryTag);
-        table.clear();
+        if (!table.isInventoryEmpty()) {
```

Comment:
Guess you don't normally create more TinkerTables than you want to place, so it's usually a stack of 1 anyway :)